core/src/main/java/hudson/init/InitMilestone.java

```diff
@@ -82,13 +85,29 @@
      * By this milestone, all programmatically constructed extension point implementations
      * should be added.
      */
-    EXTENSIONS_AUGMENTED("Augmented all extensions"), // TODO nothing attains() this so when does it actually happen?
+    EXTENSIONS_AUGMENTED("Augmented all extensions"),
+
+    /**
+     * By this milestone, all the configurations are loaded from file system
+     */
+    SYSTEM_CONFIG_LOADED("System config loaded"),
+
+    /**
+     * By this milestone, the system configuration is adapted just in case any plugin (CasC might be an example) needs
+     * to update configuration files
+     */
+    SYSTEM_CONFIG_ADAPTED("System config adapted"),
```

Comment:
OT but since we are touching this, do we want to make sure / document groovy init hooks are called here?

Reply:
Disregard, I see you did that. Great

Reply:
Later we can introduce more Goorvy hooks if needed (e.g. before the jobs load)